### DISCUSSION TITLE
Can't completely disable github

### DISCUSSION THREAD

**Discussion by rcdailey** [semantic-release/semantic-release#2955] [Configuration] [answered]
I'm new to semantic-version. I'm trying to determine if it's the right tool for some requirements I have. In my playground repo, I installed semantic-release as instructed in the docs. I noticed by default it integrates with github. I'm using ADO to host my git repo, so I needed to turn off github completely.

Based on my research, it seemed like to disable the `@semantic-release/github` plugin, I needed to specify my own `.releaserc.yml` file which I have with the following:

```yml
branches:
  - master

plugins:
  - "@semantic-release/commit-analyzer"
```

However, when I run `npx semantic-release`, I still get this prompt (I'm on Windows 10):

![image](https://github.com/semantic-release/semantic-release/assets/1768054/31b2f9c5-6800-41c5-856f-405e171eb5ea)

And if I exit this modal, I get an error:

```txt
Error: Command failed with exit code 128: git ls-remote --heads https://github.com/n/a.git
fatal: helper error (-1): User cancelled dialog.
bash: line 1: /dev/tty: No such device or address
error: failed to execute prompt script (exit code 1)
fatal: could not read Username for 'https://github.com': No such file or directory
    at makeError (file:///C:/code/Monitor/node_modules/semantic-release/node_modules/execa/lib/error.js:60:11)
    at handlePromise (file:///C:/code/Monitor/node_modules/semantic-release/node_modules/execa/index.js:124:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async getBranches (file:///C:/code/Monitor/node_modules/semantic-release/lib/git.js:68:11)
    at async default (file:///C:/code/Monitor/node_modules/semantic-release/lib/branches/expand.js:6:23)
    at async default (file:///C:/code/Monitor/node_modules/semantic-release/lib/branches/index.js:14:26)
    at async run (file:///C:/code/Monitor/node_modules/semantic-release/index.js:68:22)
    at async Module.default (file:///C:/code/Monitor/node_modules/semantic-release/index.js:275:22)
    at async default (file:///C:/code/Monitor/node_modules/semantic-release/cli.js:55:5) {
  shortMessage: 'Command failed with exit code 128: git ls-remote --heads https://github.com/n/a.git',
  command: 'git ls-remote --heads https://github.com/n/a.git',
  escapedCommand: 'git ls-remote --heads "https://github.com/n/a.git"',
  exitCode: 128,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: 'fatal: helper error (-1): User cancelled dialog.\n' +
    'bash: line 1: /dev/tty: No such device or address\n' +
    'error: failed to execute prompt script (exit code 1)\n' +
    "fatal: could not read Username for 'https://github.com': No such file or directory",
  cwd: 'C:\\code\\Monitor',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
```

I'm a bit confused why it's trying to run this command:

```txt
git ls-remote --heads "https://github.com/n/a.git"
```

I disabled the github plugin, so where is this coming from?

Due diligence performed:

- I've reviewed the [FAQ](https://semantic-release.gitbook.io/semantic-release/support/faq)
- I've searched existing discussions and issues (both open and closed).
- Google and stackoverflow search 

Let me know if there's additional information I may provide. Thank you in advance for the help!

**Comment by travi** [comment, marked as answer]
do you happen to have a `repository` property in your `package.json` that points to github?

**Reply by rcdailey** [reply]
Thank you so much for the quick reply!

Yes, in fact, it seems that I do have that:

```json
"repository": {
  "type": "Perforce",
  "url": "n/a"
},
```

I'm not sure how this got put here. Should I completely remove this section?

**Reply by almcaffee** [reply]
I am having this issue without this repository property.

**Reply by babblebey** [reply]
Do you have a .releaserc config?

**Reply by zmeggyesi** [reply]
I'm having pretty much the same issue: I have a `repository` object in my `package.json` - which I should because I want the GH to be discoverable via NPM's interfaces - but I don't want the GitHub plugin to do anything because the release process doesn't work that way.

How do I **actually** disable the plugin, so that it doesn't try to silently use the repo from my package file?

**Reply by travi** [reply]
as long as the `repository` definition in the `package.json` matches the real remote, things should work fine. the example above was broken because the `url` was not set to match the real remote (and I don't have direct experience with Perforce to know if that would add more complications?).

if you have that set correctly, your problem is likely different. please open a new discussion

**Reply by zmeggyesi** [reply]
I will have to, since after a lot of debugging, I did realize that my problem is more of a feature request than it's a bug. I'll need to spend some more time to word it correctly, though.
